JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -12,6 +13,7 @@ namespace JustSaying.AwsTools.MessageHandling
     public abstract class SnsTopicBase : IMessagePublisher
     {
         private readonly IMessageSerialisationRegister _serialisationRegister; // ToDo: Grrr...why is this here even. GET OUT!
+        public SnsWriteConfiguration SnsWriteConfiguration { private get; set; }
```

Comment:
Public set, private get seems a bit weird to me. Why not just make them both public?

Reply:
Was just about to make the same comment. I'd like this to be set only on construction, if that means adding an overload to the constructor and having a field then so be it.

Reply:
tbf it doesn't help that this class is public when it should really be internal. I have a branch where I go nuts making things internal, which can wait for v6

Reply:
No, that's a fair point @martincostello. I'll update it.

Reply:
Wasn't sure about constructor injection when I did it but in hindsight it's a good call. Update incoming.